src/modding/ModConfigItemInfo.cs

```diff
@@ -0,0 +1,214 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using Godot;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+/// <summary>
+///   Class that holds the info of a mod config item from a 'mod_config.json' file
+/// </summary>
+public class ModConfigItemInfo : HBoxContainer
+{
+    public string? ID { get; set; }
+
+    [JsonProperty("Display Name")]
+    public string? DisplayName { get; set; }
+
+    public string? Description { get; set; }
+
+    public string? Type { get; set; }
+
+    // Exclusively for Enums/Option type
+    public object? Options { get; set; }
+
+    public object? Value { get; set; }
+
+    [JsonProperty("Min")]
+    public float MinimumValue { get; set; }
+
+    [JsonProperty("Max")]
+    public float MaximumValue { get; set; } = 99f;
+
+    public override bool Equals(object other)
+    {
+        var item = other as ModConfigItemInfo;
+
+        if (item == null)
+        {
+            return false;
+        }
+
+        return ID == item.ID && Type == item.Type;
+    }
+
+    /// <summary>
+    ///   Updates the Value variable based on the gui element
+    /// </summary>
+    public object UpdateInternalValue()
+    {
+        var nodeChildren = GetChildren();
+        if (nodeChildren.Count > 1)
+        {
+            var childUIElement = GetChild(1);
+            switch (Type?.ToLower(CultureInfo.CurrentCulture) ?? string.Empty)
+            {
+                case "int":
+                case "integer":
+                case "i":
```

Comment:
I'm not super fan of such variations in the code, since it tends to create many variants of a single code and may end up being confusing.

Reply:
This seems to be the place that's related to my previous comment.

Reply:
i think i just keep the full name then